metadata-report is not support a separate configuration password

### Environment

* Dubbo version: 2.7.15
* Operating System version: mac os 10.15.7
* Java version: java 8

###  config

good
```bash
dubbo.metadata-report.address=redis://default:123456@11.6.0.4:6379
```
bad
```bash
dubbo.metadata-report.address=redis://11.6.0.4:6379
dubbo.metadata-report.username=default
dubbo.metadata-report.password=123456
```
error

```bash
Caused by: redis.clients.jedis.exceptions.JedisDataException: NOAUTH Authentication required.
```

###  How to Fix?

https://github.com/apache/dubbo/blob/3.0/dubbo-common/src/main/java/org/apache/dubbo/config/MetadataReportConfig.java

####  current
![image](https://user-images.githubusercontent.com/28058794/162732942-109288e0-8db0-4bf5-af11-31003f7629be.png)

####  After Fix
```java
    public URL toUrl() throws IllegalArgumentException {
        String address = this.getAddress();
        if (isEmpty(address)) {
            throw new IllegalArgumentException("The address of metadata report is invalid.");
        }
        Map<String, String> map = new HashMap<String, String>();
        URL url = URL.valueOf(address);
        // Issue : https://github.com/apache/dubbo/issues/6491
        // Append the parameters from address
        map.putAll(url.getParameters());
        // Append or overrides the properties as parameters
        appendParameters(map, this);
        // Normalize the parameters
        map.putAll(convert(map, null));
        // put the protocol of URL as the "metadata"
        map.put("metadata", url.getProtocol());
        return new URL("metadata",
            url.getUsername() == null ? this.getUsername() : url.getUsername(),
            url.getPassword() == null ? this.getPassword() : url.getPassword(),
            url.getHost(),
            url.getPort(this.getPort()),
            url.getPath(), map);
    }
```

--- a/dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java
+++ b/dubbo-common/src/main/java/org/apache/dubbo/common/constants/CommonConstants.java
@@ -533,4 +533,6 @@ public interface CommonConstants {
     String CLEAR_FUTURE_AFTER_GET = "future.clear.once";
 
     String NATIVE_STUB = "nativestub";
+
+    String METADATA = "metadata";
 }

--- a/dubbo-common/src/main/java/org/apache/dubbo/config/MetadataReportConfig.java
+++ b/dubbo-common/src/main/java/org/apache/dubbo/config/MetadataReportConfig.java
@@ -27,6 +27,7 @@ import java.util.HashMap;
 import java.util.Map;
 
 import static org.apache.dubbo.common.constants.CommonConstants.CYCLE_REPORT_KEY;
+import static org.apache.dubbo.common.constants.CommonConstants.METADATA;
 import static org.apache.dubbo.common.constants.CommonConstants.PROTOCOL_KEY;
 import static org.apache.dubbo.common.constants.CommonConstants.REPORT_DEFINITION_KEY;
 import static org.apache.dubbo.common.constants.CommonConstants.REPORT_METADATA_KEY;
@@ -143,7 +144,7 @@ public class MetadataReportConfig extends AbstractConfig {
         if (isEmpty(address)) {
             throw new IllegalArgumentException("The address of metadata report is invalid.");
         }
-        Map<String, String> map = new HashMap<String, String>();
+        Map<String, String> map = new HashMap<>();
         URL url = URL.valueOf(address, getScopeModel());
         // Issue : https://github.com/apache/dubbo/issues/6491
         // Append the parameters from address
@@ -153,10 +154,10 @@ public class MetadataReportConfig extends AbstractConfig {
         // Normalize the parameters
         map.putAll(convert(map, null));
         // put the protocol of URL as the "metadata"
-        map.put("metadata", isEmpty(url.getProtocol()) ? map.get(PROTOCOL_KEY) : url.getProtocol());
-        return new ServiceConfigURL("metadata", url.getUsername(), url.getPassword(), url.getHost(),
-                url.getPort(), url.getPath(), map).setScopeModel(getScopeModel());
-
+        map.put(METADATA, isEmpty(url.getProtocol()) ? map.get(PROTOCOL_KEY) : url.getProtocol());
+        return new ServiceConfigURL(METADATA, StringUtils.isBlank(url.getUsername()) ? this.getUsername() : url.getUsername(),
+            StringUtils.isBlank(url.getPassword()) ? this.getPassword() : url.getPassword(), url.getHost(),
+            url.getPort(), url.getPath(), map).setScopeModel(getScopeModel());
     }
 
     public String getProtocol() {
@@ -191,7 +192,8 @@ public class MetadataReportConfig extends AbstractConfig {
                 updateParameters(params);
             } catch (Exception ignored) {
             }
-        }    }
+        }
+    }
 
     public Integer getPort() {
         return port;

--- a/dubbo-metadata/dubbo-metadata-api/src/test/java/org/apache/dubbo/metadata/report/MetadataReportInstanceTest.java
+++ b/dubbo-metadata/dubbo-metadata-api/src/test/java/org/apache/dubbo/metadata/report/MetadataReportInstanceTest.java
@@ -41,7 +41,7 @@ class MetadataReportInstanceTest {
     private MetadataReportConfig metadataReportConfig;
     private ConfigManager configManager;
 
-    private String registryId = "9103";
+    private final String registryId = "9103";
 
     @BeforeEach
     public void setUp() {
@@ -52,6 +52,9 @@ class MetadataReportInstanceTest {
 
         URL url = URL.valueOf("metadata://127.0.0.1:20880/TestService?version=1.0.0&metadata=JTest");
         metadataReportConfig = mock(MetadataReportConfig.class);
+        when(metadataReportConfig.getUsername()).thenReturn("username");
+        when(metadataReportConfig.getPassword()).thenReturn("password");
+
         when(metadataReportConfig.getApplicationModel()).thenReturn(applicationModel);
         when(metadataReportConfig.toUrl()).thenReturn(url);
         when(metadataReportConfig.getScopeModel()).thenReturn(applicationModel);
@@ -65,8 +68,7 @@ class MetadataReportInstanceTest {
 
     @Test
     public void test() {
-        Assertions.assertNull(metadataReportInstance.getMetadataReport(registryId),
-            "the metadata report was not initialized.");
+        Assertions.assertNull(metadataReportInstance.getMetadataReport(registryId), "the metadata report was not initialized.");
         assertThat(metadataReportInstance.getMetadataReports(true), Matchers.anEmptyMap());
 
         metadataReportInstance.init(Arrays.asList(metadataReportConfig));
@@ -79,6 +81,9 @@ class MetadataReportInstanceTest {
         Map<String, MetadataReport> metadataReports = metadataReportInstance.getMetadataReports(true);
         Assertions.assertEquals(metadataReports.size(), 1);
         Assertions.assertEquals(metadataReports.get(registryId), metadataReport);
+
+        Assertions.assertEquals(metadataReportConfig.getUsername(), "username");
+        Assertions.assertEquals(metadataReportConfig.getPassword(), "password");
     }
 
 }

